src/Datadog.Trace/Util/DomainMetadata.cs

```diff
@@ -100,12 +101,15 @@ public static int AppDomainId
 
         public static bool ShouldAvoidAppDomain()
```

Comment:
@lucaspimentel FOr my education - why are we checking for AppInsights here?

Reply:
In azure app services there are AppInsights AppDomains which spin up underneath the web worker and those would either break or submit traces for AppInsights.
It is a best effort at harmonious co-existence.